internal/shards/inmemory_repository.go

```diff
@@ -34,9 +34,12 @@ func NewMockRepository() *InMemoryRepository {
 }
 
 func (r *InMemoryRepository) Lookup(shardKey string) (string, error) {
-	if shard, exists := r.Shards[shardKey]; exists {
-		return shard.ShardName, nil
+	for _, mapping := range r.Shards {
+		if mapping.ShardKey == shardKey {
+			return mapping.ShardName, nil
+		}
 	}
+
```

Comment:
Given a config file like the following:

```yaml
mappings:
  foo1:
    shardKey: foo2
    shardName: live-txns
  foo2:
    shardKey: foo1
    shardName: fake-txns
```

`.Lookup("foo1") == "live-txns"` would be true. Now it would be `.Lookup("foo1") == "fake-txns"`.

It seems like we should cleanup this messy config rather than patch the behavior.